units/provisioning_create_host.go

```diff
@@ -69,8 +69,6 @@ func NewHostCreateJob(env evergreen.Environment, h host.Host, id string, current
 		j.MaxAttempts = maxAttempts
```

Comment:
It looks like this value used to be set to 1, for a total of 2 attempts because it's zero indexed. Now, we're getting 1 attempt, with no retries, because you're calling NewHostCreateJob with maxAttempts = 0, which should mean this block is not entered and j.MaxAttempts is the struct.int default of 0. So we don't get a default retry through tryRequeue, but we're getting that retry through the ch.Retries above, and consequently we don't need it here?

Reply:
The value I changed was for currentAttempt, not maxAttempts. NewHostCreateJob was always maxAttempts = 0, I changed it so that currentAttempt also started at 0. When we retry, we use the passed in j.MaxAttempts, so we would hit this block. 
However, we could probably assume that j.host.SpawnOptions.MaxAttempts will always be set to something, since we do set it to something when we validate SpawnOptions (I changed this default to be 2 attempts), and then we don't have to bother passing around maxAttempts. I can change it to that for clarity.

Reply:
OK, that makes sense.